mozilla/addons-server
# rm sponsored, verified

Fixes: mozilla/addons#14996


### Description

Removes sponsored and verified promoted groups.  Also add an API gate to silently ignore those promoted group names if passed as a search parameter - currently enabled for v3, v4, and v5, but we could drop v5 once frontend is fixed (as it's explicitly positioned as unstable).

### Context

Frontend (and probably autograph) would be separate PRs.

### Testing

- use django admin to try to create a promoted addon; see Verified and Sponsored are no longer available
- make a search API call with a `?promoted=sponsored` param
  - see no results but no error
  - repeat with `verified`
  - repeat with other promoted groups both separately and with verified/sponsored (it's a comma-separated list)

### Checklist
